packages/core/README.md

```diff
@@ -23,7 +23,357 @@ npm install @vue-kakuyaku/core
+
+Luckily for us, Vue provides [API](https://github.com/vuejs/rfcs/blob/master/active-rfcs/0041-reactivity-effect-scope.md) for creating our own scopes! With them, we can isolate async actions and their reactive stuff within a dedicated scope which we can set up and dispose at any time. `useDeferredScope<T>()` does exactly this.
+
+See how it works by example:
+
+```ts
+interface Params {
+  username: string
+  password: string
+}
+
+const {
+  // it is a `Ref<null | { expose: T }>, where `expose` is
+  // what is returned from the scope's setup function
+  scope: loginScope,
+
+  setup: loginSetup,
+  dispose: cleanLogin,
+} = useDeferredScope<{
+  isOk: Ref<boolean>
+  retry: () => void
+  // sometimes we want to know the exact params of the last login
+  params: Params
+}>()
+
+function doLogin(params: Params) {
+  // if the scope is already set up, it will be disposed
+  loginSetup(() => {
+    // during this function we can setup any reactive logic and
+    // be calm that it will be cleared automatically on scope dispose
+
+    const { state, run } = useTask(() => httpLogin(params), {
+      immediate: true,
+    })
+
+    const isOk = computed(() => !!state.fulfilled)
+
+    return { isOk, params, retry: run }
+  })
+}
+
+const isLoginOk = computed(
+  () => loginScope.value?.expose.isOk ?? false,
+)
+
+function retryLogin() {
+  loginScope.value?.expose.retry()
+}
+```
+
+> We use `expose` keyword, trying to follow [existing Vue semantics](https://vuejs.org/api/options-state.html#expose) around this keyword.
+
+This utility is a bit low-level and not very useful for a very common case - when you need to set up an async action, based on reactive parameters. The next section gives a solution for it.
+
+### Reactively parametrised scope with `useParamScope()`
+
+Such a common scenario: use have a reactive `userId` and you need to fetch a user data, according to the ID. Additionally, you might need to set up some reactive logic around it.
+
+Here is how it might look with `useParamScope()`:
+
+```ts
+const userId = ref(1)
+
+const scope = useParamScope(
+  // reactive key - a ref or a getter
+  userId,
+
+  // setup function, that accepts the resolved key for the scope
+  (staticUserId) => {
+    const { state, run } = useTask(
+      () => fetch(`/users/${staticUserId}`),
+      { immediate: true },
+    )
+
+    useErrorRetry(state, run)
+    const staleState = useStaleState(state)
+
+    return staleState
+  },
+)
+
+const userData = computed(() => scope.value.expose.fulfilled?.value)
+```
+
+Whenever the reactive (and **primitive**, i.e. `number | string | symbol | boolean`, such as [Vue's `key` attr](https://vuejs.org/api/built-in-special-attributes.html#key) (except `boolean`)) key is changed, the existing scope is disposed, and the new one is set up.
```

Comment:
```suggestion
Whenever a reactive key is changed, the existing scope is disposed, and the new one is set up. The same applies for **primitive** keys: `number | string | symbol | boolean`.
```


not sure what this part means:
```
such as [Vue's `key` attr](https://vuejs.org/api/built-in-special-attributes.html#key) (except `boolean`))
```

Reply:
"The same applies..." doesn't seem correct.

I meant that the reactive *key* should be *primitive* anyway.

As for Vue: https://vuejs.org/guide/essentials/list.html#maintaining-state-with-key

> The `key` binding expects primitive values - i.e. strings and numbers. Do not use objects as v-for keys.

```html
<div v-for="i in 5" :key="i"">{{ i }}</div>
```

Vue developers know good about the `key` special attribute that Vue uses to distinguish elements from each other. Only primitive value should be bound to it, and the same rule applies for our keys.

Reply:
```suggestion
Whenever a reactive key is changed, the existing scope is disposed, and the new one is set up. The reactive keys should be **primitive** keys: `number | string | symbol | boolean`.
```